test: random perf latency mode

psavol opened the pull request:
Added new -m/-d options into random number performance test application, which are used to test latency of random number generation.

TuomasTaipale commented on test/performance/odp_random.c:
```diff
@@ -14,8 +14,32 @@
+	test_global_t *global;
+	int thread_idx;
+	uint8_t *data;
+
```
Extra newline.

psavol replied:
I prefer this style where the name of the type/struct is easier to find after the list of struct members.

TuomasTaipale replied:
OK, maybe same treatment to `options_t`.

psavol replied:
Done in v2.

TuomasTaipale commented on test/performance/odp_random.c:
```diff
@@ -97,198 +121,145 @@ static int parse_options(int argc, char *argv[])
+	uint32_t rounds = global->rounds;
+
+	/* One warm up round */
+	if ((int)type == PSEUDO_RANDOM)
+		pseudo_random_data_loop(1, data, size);
+	else
+		random_data_loop(type, 1, data, size);
```
This seems to repeat, could be in a separate function or alternatively `*_data_loop` signatures could be unified and called via a pointer.

psavol replied:
I think a new function is not necessary in this case, as it would save so few lines (add 6 new lines, remove 9 lines).

TuomasTaipale replied:
OK. In general, it's a bit cumbersome to require the type check everytime there's a call to get random data.

psavol replied:
Fixed in v2. Moved pseudo_random_data_loop() inside random_data_loop() as it had all needed arguments already.

TuomasTaipale commented on test/performance/odp_random.c:
```diff
@@ -200,6 +237,97 @@ static int test_random_perf(void *ptr)
+
+		if ((int)type == PSEUDO_RANDOM) {
+			t1 = odp_time_local_strict();
+			while (pos < size) {
```
These inner blocks seem very similar to the `*_data_loop` functions, reuse those?

psavol replied:
The difference is time measurement: for latency it is measured for every round, for performance it is measured once for all rounds (to avoid measurement overhead).

TuomasTaipale reviewed (commented):
Typo in 5bae514 "Added also a *wram* up".

psavol commented:
> Typo in [5bae514](https://github.com/OpenDataPlane/odp/commit/5bae514601098623c7c1c7bf596b0c4f7289ef7e) "... Added also a _wram_ up ...".

Fixed in v2
